### PR TITLE
Make ProcessManagerDispatcher.dispatch return generated events

### PR DESCRIPTION
**Pull request by vlad-lubenskyi**
Prior to this PR, it was impossible to test if Process Manager generates any events as a reaction to another event.

The issue was that `ProcessManagerDispatcher.dispatch(ProcessManager, EventEnvelope)` ignored messages returned by `ProcessManager.dispatchEvent(...)`.

This pull request:

1. Makes `ProcessManagerDispatcher.dispatch(ProcessManager, EventEnvelope)` return list of produced events.
2. Updates Javadoc respectively.
3. Updates unit test respectively.

### CODE REVIEW

**Comment by vlad-lubenskyi** [comment]
@armiol PTAL

File: testutil-server/src/main/java/io/spine/server/procman/ProcessManagerDispatcher.java [hunk 1]
```diff
@@ -58,14 +58,19 @@ private ProcessManagerDispatcher() {}
 
     /**
      * Dispatches an {@linkplain EventEnvelope event} to the given {@code ProcessManager}.
+     *
+     * @return the list of event messages generated by Process Manager, or empty list if none were
```
Comment (armiol) [inline testutil-server/src/main/java/io/spine/server/procman/ProcessManagerDispatcher.java]: I would say `... generated by the given {@code ProcessManager}...`

File: testutil-server/src/main/java/io/spine/server/procman/ProcessManagerDispatcher.java [hunk 2]
```diff
@@ -58,14 +58,19 @@ private ProcessManagerDispatcher() {}
...
+     * generated.
      */
-    public static void dispatch(ProcessManager<?, ?, ?> pm, EventEnvelope event) {
+    public static List<Event> dispatch(ProcessManager<?, ?, ?> pm, EventEnvelope event) {
         checkNotNull(pm);
         checkNotNull(event);
 
         final PmTransaction<?, ?, ?> tx = PmTransaction.start(pm);
-        pm.dispatchEvent(event);
+        List<Event> eventMessages = pm.dispatchEvent(event);
```
Comment (armiol) [inline testutil-server/src/main/java/io/spine/server/procman/ProcessManagerDispatcher.java]: This variable looks `final`.

File: server/src/test/java/io/spine/server/procman/ProcessManagerShould.java [hunk 3]
```diff
@@ -122,7 +123,11 @@ public void have_default_state_initially() {
 
     @Test
     public void dispatch_event() {
-        testDispatchEvent(Sample.messageOfType(PmProjectStarted.class));
+        final List<? extends Message> eventMessages = testDispatchEvent(
```
Comment (armiol) [inline server/src/test/java/io/spine/server/procman/ProcessManagerShould.java]: I would format it like this:
```
final List<? extends Message> eventMessages = 
         testDispatchEvent(Sample.messageOfType(PmProjectStarted.class));
```

**Comment by vlad-lubenskyi** [comment]
@armiol PTAL

**Pull request merged**